utils/handleScroll.ts

```diff
@@ -0,0 +1,31 @@
+import type { Ref } from 'vue'
+
+export interface SectionInformation {
+    sectionTitle: string
+    sectionElementIdToScrollTo: string
+}
+
+export const handleScroll
+= (isScrolling: Ref<boolean>, activeSection: Ref<string>, sectionDetailsObject: SectionInformation[]) => {
+    if (isScrolling.value) return
+    let newActiveSection: string | null = null
+
+    sectionDetailsObject.forEach(section => {
+        let rect = null
+        const foundSectionById = document.getElementById(section.sectionElementIdToScrollTo)
+        if (foundSectionById) {
```

Comment:
minor: use return-early ifs format. If you invert this `if`, then you can remove the `else`

Reply:
this one isn't resolved 😆